3.11 upgrade updates

kalexand-rh opened the pull request:
Updates to the upgrade topics for 3.11. This includes my understanding of the changes for https://trello.com/c/A6B7w2dE/965-document-check-for-expiring-certificates-during-upgrade

kalexand-rh commented on install/configuring_inventory_file.adoc:
```diff
@@ -182,6 +182,16 @@ for the hosted router.
 validity for etcd CA, peer, server and client certificates. Defaults to `1825`
 (5 years).
 
+|`openshift_certificate_expiry_warning_days`
```
This is directly related to the Trello card.

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -326,102 +181,33 @@ the host you run the upgrade playbooks on:
 run properly or other cluster stability issues.
-+
-In particular, if you made any changes to `admissionConfig` settings in your
+.. To check if your certificates will expire, set the
```
This is directly related to the Trello card.

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -636,255 +405,170 @@ openshift_node_upgrade_post_hook=/usr/share/custom/post_node.yml
-== Special Considerations When Using Containerized GlusterFS
-When upgrading  {product-title}, you must upgrade the set of nodes where
-GlusterFS pods are running.
+== Upgrading {product-title} when using containerized GlusterFS
```
Do these instructions still work? I'd like to update this section to explain which playbooks to run, but I am not sure how to upgrade the masters, then the infrastructure, then the individual nodes. Would you run a control plane upgrade and then manually upgrade each node in the cluster?

kalexand-rh commented:
@vikram-redhat. I'm having trouble tracking down this build error. If you have time, will you PTAL?

kalexand-rh commented:
@xltian, @gpei will you PTAL? I've included both general upgrade updates and the update to [check for expiring certificates during upgrade](https://trello.com/c/A6B7w2dE/965-document-check-for-expiring-certificates-during-upgrade).

jianlinliu commented:
@wmengRH @jiajliu Help review upgrade part.

gpei commented:
The update to `check for expiring certificates during upgrade` lgtm, thanks

jiajliu commented:
@kalexand-rh 
It seems I can not read the full context in section [in-place upgrade]. (https://github.com/openshift/openshift-docs/blob/1e2b280bbc3fb4788c3060f82161cd5f6782bf18/upgrading/automated_upgrades.adoc)
And seems it include some origin context in this pr. Could u give a link for review this section of ocp?

jianlinliu commented:
@prasanth-RH pls help involve cns qe for the 3.11 glusterfs upgrade doc review.

jiajliu commented:
@anpingli @juzhao @gpei @jianlinliu 
Please help review your own parts if any changes.

jianlinliu commented:
Indeed, we need some pre-review for better reading.

kalexand-rh commented:
@vikram-redhat, is there a good way to force the preview bot to build a PR?

@jianlinliu, @jiajliu, here's a build preview. I'll update it with more changes: http://file.rdu.redhat.com/kalexand/092018/3.11_updates/upgrading/automated_upgrades.html

jianlinliu commented on upgrading/automated_upgrades.adoc:
```diff
@@ -1002,7 +748,7 @@ master-etcd-master1.example.com          1/1       Running   6          5d
 
 . Verify that you are running the expected versions of the *docker-registry*
-and *router* images, if deployed.
+and *router* images, if deployed:
```
Seeing from the pre-review draft, the following check command is still using v3.10 image tag, should be updated to v3.11.

jianlinliu replied:
Also the output of `oc get nodes` is showing v1.10, should be some v1.11 string.

kalexand-rh replied:
@jianlinliu, I'm adding  a v.1.11 string, though I'll be happy to change it if you have a better value, and updated the image tag. Please let me know if you hear what the latest version tag will be.

jianlinliu replied:
@kalexand-rh The first version of 3.11 image tag would be v3.11.16, and `oc get nodes` is showing "v1.11.0+d4cacc0"

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -636,255 +407,169 @@ openshift_node_upgrade_post_hook=/usr/share/custom/post_node.yml
+Before you upgrade a mixed environment, for example one with Red Hat Enterprise
+Linux (RHEL) and RHEL Atomic Host, you must set values in your inventory file
+for both the `openshift_pkg_version` and `openshift_image_tag` parameter. In mixed
+environments, if you specify only `openshift_pkg_version`, then that number is
```
@jianlinliu, this is existing text that I moved. It was here: https://github.com/openshift/openshift-docs/blame/master/upgrading/automated_upgrades.adoc#L864 and came from this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1513693

Is it still true that if you don't specify a value for `openshift_image_tag`, the installer uses the value that you specified for `openshift_pkg_version`?

kalexand-rh replied:
@jianlinliu, please let me know if you want more changes here.

jianlinliu replied:
@wmengRH @jiajliu could you help review and give a confirm here?

jianlinliu replied:
@kalexand-rh after review openshift-ansible code, when I almost confirm the present description is not accurate. When `openshift_image_tag` and `openshift_pkg_version` is not defined in inventory file, they would be aligned with `openshift_release` whose default value is 3.11. There is no direct connection between `openshift_image_tag` and `openshift_pkg_version`. So I am proposing this:
"you must set values" -> "It is better to set values"
"In mixed environments, if you specify ... <to end>" -> "That would assure your cluster is running the same version of openshift, whatever the node is a RHEL or RHAH"

kalexand-rh commented:
I updated the files at the preview link: http://file.rdu.redhat.com/kalexand/092018/3.11_updates/upgrading/automated_upgrades.html

gpei commented on upgrading/automated_upgrades.adoc:
```diff
@@ -326,102 +181,33 @@ the host you run the upgrade playbooks on:
+`openshift_certificate_expiry_warning_days` parameter value to the number of
+days that your certificate must valid. For example, to ensure that your
+certificates are good for a year, set
+`openshift_certificate_expiry_warning_days` to `365`.
```
We could also add the usage of `openshift_certificate_expiry_warning_days` parameter here. 
Set openshift_certificate_expiry_warning_days to False could bypass the check.

gpei commented on upgrading/automated_upgrades.adoc:
```diff
@@ -639,255 +410,169 @@ openshift_node_upgrade_post_hook=/usr/share/custom/post_node.yml
 
 [[upgrade-hooks-nodes]]
-.Node Upgrade Hooks
+.Node upgrade hooks
```
One known issue about the node upgrade hooks - [BZ#1624799](https://bugzilla.redhat.com/show_bug.cgi?id=1624799) , it's worth mentioning here IMO.

kalexand-rh replied:
I added a limitation for this one.

kalexand-rh commented:
@juzhao, will you double check? I agree that those parameters aren't related to metrics, but I think that might be ok. You're using them to set up storage for metrics, not the metrics themselves.

kalexand-rh commented:
@gpei, will you please take a look at how I addressed your comments on the most recent commit?

jiajliu commented on upgrading/automated_upgrades.adoc:
```diff
@@ -20,300 +20,155 @@ endif::[]
 ----
 $ oc adm migrate storage --include=* --loglevel=2 --confirm --config /etc/origin/master/admin.kubeconfig
 ----
 
```
For v3.10 to v3.11 upgrade, there should be a matched node for Cluster Monitoring Operator which would be deployed by default. Or else, the upgrade will be broken due to no scheduled nodes found matching node selector for Cluster Monitoring Operator. So before upgrade, another prerequisite is that ensure you have infra node in the cluster(infra node is the default node for Cluster Monitoring Operator), if there is not infra node or user want to schedule it on other node, they need to specify the targeted node with openshift_cluster_monitoring_operator_node_selector, or if user did not want it deployed, then they need to disable it by specifying openshift_cluster_monitoring_operator_install=false.

jiajliu replied:
or you can add this description in following section "Review and update your inventory file."

jiajliu commented:
@stuartchuan 
Please take care of part "Updating policy definitions".

jiajliu commented:
@kalexand-rh 
Another important thins is about default registry will be changed to registry.redhat.io during upgrade if no customized registry specified. And the new registry need specify credentials in inventory file. After upgrade, all images should be pulled from new registry except ose-pod image.

gpei commented on upgrading/automated_upgrades.adoc:
```diff
@@ -368,9 +371,11 @@ warning that your hook needs to be updated to work with the latest
 process. If you get an error, you must address the problem and then start the
 upgrade again.
+- You can run upgrade hooks on only nodes, not masters. To run the hooks on
+masters, you must specify a master hook for those nodes. 
```
To be more precise, we could make it as   "You can run `node upgrade hooks` on only nodes, not masters. "

gpei commented on upgrading/automated_upgrades.adoc:
```diff
@@ -192,11 +192,14 @@ changes you made, apply the equivalent appropriate changes to your inventory
 run properly or other cluster stability issues.
-.. To check if your certificates will expire, set the
+.. To check if your certificates will expire, set
+`openshift_certificate_expiry_fail_on_warn` to `True` and the
```
We don't need this explain IMHO. This parameter default to be True.

kalexand-rh replied:
@gpei, if they're upgrading from 3.10, would that parameter be in the file?

stuartchuan commented:
"Updating policy definitions" part is OK.

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -326,102 +181,47 @@ the host you run the upgrade playbooks on:
-made the equivalent changes to your inventory file, review
-xref:../install/configuring_inventory_file.adoc#configuring-ansible[Configuring
-Your Inventory File]. For any variables that are relevant to the manual
+.. Prepare for the Cluster Monitoring Operator. In version 3.11, the Cluster
```
@jiajliu, does this sound ok?

jiajliu replied:
lgtm now

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -326,102 +181,47 @@ the host you run the upgrade playbooks on:
 run properly or other cluster stability issues.
-+
-In particular, if you made any changes to `admissionConfig` settings in your
+.. By default, the installer checks to see if your certificates will expire
```
@gpei, is this better?

gpei replied:
yeah, lgtm

wmengRH commented:
@kalexand-rh  this should be updated in 3.11 as well, thanks. 
https://bugzilla.redhat.com/show_bug.cgi?id=1627671

jiajliu commented on upgrading/automated_upgrades.adoc:
```diff
@@ -20,300 +20,155 @@ endif::[]
 +
 ----
 /usr/lib/systemd/system/atomic-openshift-*.service
 /etc/origin/node/node-config.yaml
```
There is not /usr/lib/systemd/system/atomic-openshift-*.service in v3.10.

jiajliu replied:
@kalexand-rh 
Seems this comment did not update in the pr.

kalexand-rh replied:
@jiajliu, "/usr/lib/systemd/system/atomic-openshift-*.service" in not in the file in master.

kalexand-rh commented:
@jiajliu, @bmcelvee is working on the registry change on https://github.com/openshift/openshift-docs/pull/11506/files. I'm adding a step to the upgrade topic that links to the install_config topic.

@wmengRH, I'll take that bug and address it soon.

kalexand-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -193,6 +197,11 @@ an infrastructure node to your cluster.
 xref:../admin_guide/manage_nodes.adoc#updating-labels-on-nodes[marking it]
 with the `openshift_cluster_monitoring_operator_node_selector`.
 
+.. If you use the standard {product-title} registry, prepare for the change from
```
@jiajliu, does this sound ok for the registry change?

jiajliu replied:
@kalexand-rh 
This change seems not updated in the pr. Could u show me the new open pr for it.

kalexand-rh replied:
It's in the PR but not in the preview because I did not manually update it first. I made a new PR here for follow-up requests: https://github.com/openshift/openshift-docs/pull/12355

The change should be there.

kalexand-rh commented:
@jiajliu, @gpei, @jianlinliu, @juzhao, @xltian

@vikram-redhat asked me to go ahead and merge these changes. I think I've addressed all your feedback so far. If you have more changes or I missed something, please tag me in new comments on the PR so I can open a new one or make a bug and assign it to me.

kalexand-rh commented:
/cherrypick enterprise-3.11

vikram-redhat commented on upgrading/automated_upgrades.adoc:
```diff
@@ -20,300 +20,159 @@ endif::[]
 
-* If you are using GCE Persistent Disk (gcePD), see xref:special-considerations-for-gcepd[Special Considerations When Using
-gcePD] before proceeding.
+* Upgrade the cluster to the
```
@kalexand-rh this section needs updating for 3.11.

kalexand-rh replied:
@vikram-rh, can you please be more specific? I thought you needed to upgrade to the latest version of 3.10 before you upgraded to 3.11.
